src/app/routes/registration/request-claim/request-claim.component.ts

```diff
@@ -178,83 +199,66 @@ export class RequestClaimComponent implements OnInit, SubjectElements {
   }
 
   async submit() {
+    if (!this.enrolmentForm.valid && !this.isRegistrationTypeSelected()) {
+      this.toastr.error('Enrolment Form is invalid.', TOASTR_HEADER);
+      return;
+    }
     this.loadingService.show();
-    if (this.enrolmentForm.valid) {
-      let did = undefined;
-      if (this.selectedRole.issuer) {
-        if (this.selectedRole.issuer.roleName) {
-          // Retrieve list of issuers by roleName
-          did = await this.iamService.iam.getRoleDIDs({
-            namespace: this.selectedRole.issuer.roleName
-          });
-        } else if (this.selectedRole.issuer.did) {
-          did = this.selectedRole.issuer.did;
-        }
-      }
-
-      if (did && did.length) {
-        this.submitting = true;
-        this.loadingService.show('Please confirm this transaction in your connected wallet.');
 
-        try {
-          // Construct Fields
-          let fields = [];
-          let values = this.enrolmentForm.value.fields;
-          for (let i = 0; i < this.fieldList.length; i++) {
-            fields.push({
-              key: this.fieldList[i].label,
-              value: values[i]
-            });
-          }
+    let issuerDids = [];
+    if (this.enrolmentForm.get('offChain').value) {
+      issuerDids = await this.getIssuerDid();
+      if (!(issuerDids && issuerDids.length > 0)) {
+        this.toastr.error('Cannot identify issuer for this role.', TOASTR_HEADER);
+        this.loadingService.hide();
+        return;
+      }
+    }
 
-          // Submit first digit of version
-          // because legacy role's had version format of '1.0.0'
-          // but we version should be persisted as numbers
-          const parseVersion = (version: string | number) => {
-            if (typeof (version) === 'string') {
-              return parseInt(version.split('.')[0], 10);
-            }
-            return version;
-          };
-          const claim = {
-            fields: JSON.parse(JSON.stringify(fields)),
-            claimType: this.selectedNamespace,
-            claimTypeVersion: parseVersion(this.selectedRole.version) || DEFAULT_CLAIM_TYPE_VERSION
-          };
+    this.submitting = true;
+    this.loadingService.show('Please confirm this transaction in your connected wallet.');
 
-          await this.iamService.iam.createClaimRequest({
-            issuer: did,
-            claim,
-            subject: this.roleTypeForm.value.assetDid ? this.roleTypeForm.value.assetDid : undefined
-          });
+    try {
+      await this.iamService.iam.createClaimRequest({
+        issuer: issuerDids,
+        claim: this.createClaim(),
+        subject: this.roleTypeForm.value.assetDid ? this.roleTypeForm.value.assetDid : undefined,
+        registrationTypes: this.getRegistrationTypes()
```

Comment:
🎉

Reply:
Well in fact I merged those two forms. Just left `isRegistrationTypeSelected` which checks if one of the checkboxes is selected. I think i can even remove this from this submit function, because when user doesn't select onchain or offchain, then submit button is disabled.

Reply:
Ah okay, right on. Why is `isRegistrationTypeSelected` not a part of `enrolmentForm.valid`? To me, intuitively, the form is only valid if one of the registration types is selected. Not a showstopper, just curious

Reply:
Yeah, i see your point. I did it like that because it was easier and faster. Right now we do not have validation for checkboxes to check if at least one is selected. I can do that if you prefer, so i will need to create new FormGroup and add new validator that checks it. 
Or easier way is to create a method `isEnrolmentFormValid` that will be used in `submit()` method and to disable submit button and this way there will be no confusion.

I think you prefer to do first approach, so i will change this.

And what i saw, that on dev environment you can send invalid enrolmentForm. I suppose it shouldn't be possible, am i right?

Reply:
Hmm, yes, it shouldn't be possible to submit invalid enrolmentForm. Is that fixed in this PR then?

Reply:
yes it is.